library(dada2)
packageVersion("dada2")
library(ShortRead)
packageVersion("ShortRead")
library(Biostrings)
packageVersion("Biostrings")
library(phyloseq)
packageVersion("phyloseq")

path <- getwd()
fnFs <- sort(list.files(".", pattern=".fastq", full.names = TRUE))
sample.names <- sapply(strsplit(basename(fnFs), ".fastq"), `[`, 1)
png(file="metagenomicsFastQC.png")
plotQualityProfile(fnFs)
dev.off()
fnFs.filtN <- file.path(path, "filtN", basename(fnFs))
filterAndTrim(fnFs, fnFs.filtN, maxN = 0, multithread = TRUE)
png(file="FiltNmetagenomicsFastQC.png")
plotQualityProfile(fnFs)
dev.off()
fnFs.filtQ <- file.path(path, "filtQ", basename(fnFs))
filterAndTrim(fnFs.filtN, fnFs.filtQ,
              maxN=0, maxEE=2, truncQ=2, rm.phix=FALSE,
              compress=TRUE, multithread=TRUE)
png(file="FiltQmetagenomicsFastQC.png")
plotQualityProfile(fnFs.filtQ)
dev.off()
dereps <- dada2:::derepFastq(fnFs.filtQ, verbose = TRUE)
err2 <- learnErrors(dereps, errorEstimationFunction=PacBioErrfun, BAND_SIZE=32, multithread=TRUE)
png(file="err2.png")
plotErrors(err2)
dev.off()
dd2 <- dada(dereps, err=err2, BAND_SIZE=32, multithread=TRUE)
st2 <- makeSequenceTable(dd2)
tax2 <- assignTaxonomy(st2, "~/dbs/silva_132.18s.99_rep_set.dada2.fa.gz", multithread=TRUE)
print(head(unname(tax2)))
unname(tax2)
bim2 <- isBimeraDenovo(st2, minFoldParentOverAbundance=3.5, multithread=TRUE)
table(bim2)
saveRDS(st2, "st2.rds")
saveRDS(tax2, "tax2.rds")
saveRDS(tax2pr2, "tax2pr2.rds")

tax2pr2 <- assignTaxonomy(st2, "~/dbs/pr2_version_4.14.0_SSU_dada2.fasta.gz", 
               taxLevels = c("Kingdom","Supergroup","Division","Class","Order","Family","Genus","Species"))
print(head(unname(tax2pr2)))
unname(tax2pr2)
